compiler/cgen.nim

```diff
@@ -539,7 +539,21 @@ proc initLocExprSingleUse(p: BProc, e: PNode, result: var TLoc) =
 
 include ccgcalls, "ccgstmts.nim"
 
-proc initFrame(p: BProc, procname, filename: Rope): Rope =
+proc initFrame(p: BProc, procname, filename: Rope): Rope =  
+  const frameDefines = """
+  $1  define nimfr_(proc, file) \
+      TFrame FR_; \
+      FR_.procname = proc; FR_.filename = file; FR_.line = 0; FR_.len = 0; #nimFrame(&FR_);
+
+  $1  define nimfrs_(proc, file, slots, length) \
+      struct {TFrame* prev;NCSTRING procname;NI line;NCSTRING filename; NI len; VarSlot s[slots];} FR_; \
+      FR_.procname = proc; FR_.filename = file; FR_.line = 0; FR_.len = length; #nimFrame((TFrame*)&FR_);
+
+  $1  define nimln_(n, file) \
+      FR_.line = n; FR_.filename = file;
+  """
+  appcg(p.module, p.module.s[cfsForwardTypes], frameDefines, [rope("#")])
```

Comment:
But this should only be added once per C file, not per proc in the C file. Am I missing something?

Reply:
oops, your are right again ;)
added extra section for frame defines